frontend/Interfaces/buddy/DIP/DIP.h

```diff
@@ -328,12 +328,15 @@ inline MemRef<float, 2> Rotate2D(Img<float, 2> *input, float angle,
 // User interface for 2D Resize.
 inline MemRef<float, 2> Resize2D(Img<float, 2> *input, INTERPOLATION_TYPE type,
                                  std::vector<float> scalingRatios) {
-  if (!scalingRatios[0] || !scalingRatios[1]) {
+  if (scalingRatios[0] <= 0 || scalingRatios[1] <= 0) {
     throw std::invalid_argument(
-        "Please enter non-zero values of scaling ratios.\n"
+        "Please enter positive values of scaling ratios.\n"
         "Note : scaling ratio = "
-        "input_image_dimension / output_image_dimension\n");
+        "output_image_dimension / input_image_dimension\n");
   }
+  std::reverse(scalingRatios.begin(), scalingRatios.end());
+  scalingRatios[0] = 1 / scalingRatios[0];
```

Comment:
I tried to do the division in benchmark code. RUN:` ./image-processing-resize-benchmark ../../benchmarks/ImageProcessing/Images/YuTu.png SCALE_FACTOR 0.33 0.35 NEAREST_NEIGHBOUR_INTERPOLATION`
The input image is [1026, 1026], but obtain OpenCV version [339, 359], Buddy version [338, 359]. 

- OpenCV: 1026   ×   0.33 = 338.58 -> 339
- Buddy: 1026 / (1/0.33) ->338

Reply:
I'll look into it and see if we can do anything to handle this

Reply:
I have pushed the fix, please have a try now.

Reply:
Now the result shape is correct.